public/scripts/experiments/estimation/estimation.js

```diff
@@ -0,0 +1,465 @@
+                            .attr("height", new_radius);
+                    }
+                });
+        }
+
+    }
+
+    /**
+     * D3 code for plotting a triangle.
+     *
+     * @param  chart      {svg object}
+     *         radius     {double}
+     *         max_radius {double}     Largest radius of the given trial
+     *         diff       {double}     Difference between max and min radius of given trial
+     */
+    plot_triangle(chart, radius, y_pos, x_pos, is_ref) {
+        let exp = this;
+        let poly = [
+            {"x":x_pos, "y":(-0.5 * radius + y_pos)},
+            {"x":(-0.5 * radius + x_pos), "y":(0.5 * radius + y_pos)},
+            {"x":(0.5 * radius + x_pos), "y":(0.5 * radius + y_pos)}];
+
+        chart.selectAll("polygon")
+            .data([poly])
+            .enter().append("polygon")
+            .attr("points",function(d) {
+                return d.map(function(d) { return [d.x, d.y].join(","); }).join(" ");})
+            .attr("fill", exp.curr_trial_data.fill_color)
+            .attr("id", "triangle_shape");
+
+        if (is_ref === false) {
+            d3.select("body")
+                .on("keydown", function () {
+                    let event = d3.event;
+                    if (event.key === "m" || event.key === "z") {
+                        let sign = event.key === "m" ? 1 : -1;
+                        let change = Math.random() * exp.PIXEL_TO_CM * exp.MAX_STEP_SIZE;
+                        let new_radius = radius + sign * change;
+                        radius = new_radius;
+                        poly = [
+                            {"x":(x_pos), "y":(-0.5 * new_radius + y_pos)},
+                            {"x":(-0.5 * new_radius + x_pos), "y":(0.5 * new_radius + y_pos)},
+                            {"x":(0.5 * new_radius + x_pos), "y":(0.5 * new_radius + y_pos)}];
+
+                        exp.curr_trial_data.adjustments.push(change * sign / exp.PIXEL_TO_CM );
+                        exp.curr_trial_data.estimated_size = new_radius / exp.PIXEL_TO_CM ;
+                        chart.selectAll("polygon").remove();
+                        chart.selectAll("polygon")
+                            .data([poly])
+                            .enter().append("polygon")
+                            .attr("points",function(d) {
+                                return d.map(function(d) { return [d.x, d.y].join(","); }).join(" ");})
+                            .attr("fill", exp.curr_trial_data.fill_color)
+                            .attr("id", "triangle_shape");
+                    }
+                });
+        }
+    }
+
+    /*
+    * Saves experiment data as csv
+    * */
+    export_trial_data() {
+        let trial_data = jsPsych.data.get().filterCustom(function (row) {
+            return row.block_type === "practice" || row.block_type === "test";
+        })
+            // These are variables forced on by jsPsych
+            .ignore('stimulus')
```

Comment:
Might also need to be tracking number of adjustments per round - I know they wanted this for Stevens, might need to double check w/ Tina if she needs that data. 

Also saw in the export that you have an "adjustments" variable though wasn't sure what it was representing.

Reply:
the adjustments variable is an array of adjustment being made during a round. For example, [0.01, -0.01] would mean that two adjustments were made and one of the increased the radius by 0.01cm and the other one decreased the radius be 0.01cm.

So the number of adjustments being made per round is the length of the adjustments array.